raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansProcessor.java

```diff
@@ -164,9 +170,16 @@ public void init(ProcessorContext context) {
   }
 
   private boolean shouldDropSpan(TraceIdentity key, TraceState traceState) {
-    if (traceState != null
-        && maxSpanCountMap.containsKey(key.getTenantId())
-        && traceState.getSpanIds().size() >= maxSpanCountMap.get(key.getTenantId())) {
+    int inFlightSpansPerTrace =
+        traceState != null ? traceState.getSpanIds().size() : Integer.MIN_VALUE;
+    long maxSpanCountTenantLimit =
+        maxSpanCountMap.containsKey(key.getTenantId())
+            ? maxSpanCountMap.get(key.getTenantId())
+            : Long.MAX_VALUE;
+
+    if (inFlightSpansPerTrace >= defaultMaxSpanCountLimit
```

Comment:
I thought tenant specific config acts as an override the default.
But the condition here doesn't seem to indicate that. We are picking the least of both.

Reply:
What @ravisingal & I had discussed offline of having a global upper limit, and so I had prior notatin of `upper.max.span.count`. But, what you are saying is that for a specific tenant, we want a higher limit, that makes sense.

Let me know @avinashkolluru @ravisingal if we want to modify and go with that if that is making more sense.

Reply:
I'd suggest made the tenant specific config as an override. It shouldn't matter whether its is lower or higher than the default.